src/main/java/org/mitre/synthea/engine/Generator.java

```diff
@@ -307,10 +314,17 @@ public Person generatePerson(int index, long personSeed) {
         person.attributes.put(Person.LOCATION, location);
 
         LifecycleModule.birth(person, start);
+        // Force process HealthInsuranceModule in correct order.
+        if (Boolean.parseBoolean(Config.get("generate.health_insurance", "false"))) {
+          Module.processHealthInsuranceModule(person, start);
```

Comment:
It gets called elsewhere for edge cases - sometimes if an encounter occurs in a new year before the HealthInsuranceModule has a chance to run and give that person their insurance, then they will have null insurance. Which means I have to force it to process. I'll try your suggestions and see if it fixes this issue.

Reply:
I might be understanding this incorrectly but it looks like an EncounterModule gets processed at line 322 so even if the HealthInsuranceModule is first  in the list then the EncounterModule still gets called first which is what needs the health insurance to not be null. And as far as I can tell there is no way to tell where in the list the HealthInsuranceModule is.

Reply:
The only thing I can think of would be to have a method `Module.getModule(String moduleName)` that you could pass "Health Insurance Module" into that would just return the result from the retval modules map. This would at least eliminate the multiple times processHealthInsuranceModule gets called but still seems hacky in the same way.

Reply:
Fixed by removing the HealthInsuranceModule from the Module.modules map and moving it to Generator, along with the EncounterModule.